### ISSUE TITLE
Can not parse `comment on function/procedure` (postgres)

### ISSUE THREAD

**Issue by NBprojekt** [sqlfluff/sqlfluff#3561]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting the [comment on](https://www.postgresql.org/docs/current/sql-comment.html) command, sqlfluff required parentheses at the end of a function to work.
> When reading the postgres docs this parentheses are optional. 

On the other hand commenting on procedures seams not to work at all.

### Expected Behaviour

Whould be great when sqlfluff could lint functions and procedures even when the optional parentheses are missing.

### Observed Behaviour

```
Found unparsable section: 'comment
 | on procedure proc_name'
 WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
Found unparsable section: 'comment
 | on function func_name'
 WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
```

### How to reproduce

```
comment on function function1 is 'comment'; -- unparsable
comment on function function2() is 'comment'; -- success

comment on procedure procedure1 is 'comment'; -- unparsable
comment on procedure procedure2() is 'comment'; -- unparsable
```

[sqlfluff online](https://online.sqlfluff.com/fluffed?sql=H4sIAIVVxWIC_0vOz81NzStRyM9TSCvNSy7JRGIYKmQWK6gnQ1SoWyvo6iqU5hUkFhUnJuWk8nIl49FqpKGJqbm4NDk5tbiYlwtFb0FRfnJqSmlRKoJFvMVYNGO1GqEdAM8MfmXxAAAA&dialect=postgres)

### Dialect

postgres

### Version

```
$ sqlfluff --version
sqlfluff, version 1.1.0
$ python --version
Python 3.9.7
```

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rdsedmundo** [comment]
Hey @greg-finley @tunetheweb, I'm using 2.0.5 and this is still not working for me. I'm happy to open a new issue if it's appropriate, although I think just re-opening the existing one would do it, I basically have the exact same problem as reported there.

Input:
```sql
        COMMENT ON FUNCTION pubilc.test IS $$
        Lorem ipsum dolor sit amet.
        $$;
```

Expected output:
```sql
        COMMENT ON FUNCTION pubilc.test IS $$
        Lorem ipsum dolor sit amet.
        $$;
```

Actual output:
```sql
        COMMENT ON FUNCTION pubilc.testIS $$
        Lorem ipsum dolor sit amet.
        $$;
```

The `IS` parameter is getting concatened with the identifier. 

Playground link:
https://online.sqlfluff.com/fluffed?sql=H4sIAIEOO2QC_1NQgABnf19fV78QBX8_BbdQP-cQTyCjoDQpMydZryS1uETBM1hBRYWXC6pawSe_KDVXIbOguDRXISU_J79IoTizRCExN7VED6FIRcUaADPagSRfAAAA&dialect=postgres

If the identifier is changed to include `()` after the function name, it works.
